client/src/components/UserEvents/EventCard.js

```diff
@@ -41,24 +59,56 @@ export default function EventCard({ name, duration, color, link, url }) {
   const whenCopiedToClipboard = () => {
     setCopied(true);
     setTimeout(() => {
-      // such that the copied state doesn't linger forever
       setCopied(false);
     }, 3000);
   };
 
+  const handleSwitch = async () => {
+    axios
+      .put("/api/event/toggle-active", {
+        active: !eventActive,
+        eventId: _id,
+      })
+      .then((res) => console.log(res.data))
+      .catch((err) => console.log(err));
+    setEventActive(!eventActive);
```

Comment:
I wonder if it's better to put this line (`setEventActive()`) into the `.then` block during our development. That way we guarantee that the request is completed before changing the UI. If this route goes wrong, it might be hard to notice on the front end.

It's not really a pressing issue though and we can leave it as it is.

Reply:
I agree with this idea, it would be good to know if something went wrong because otherwise you could end up having the exact opposite state as the db

Reply:
I was having an async issue with setState so I did it like this but I knew it is not optimal! I can make the changes!